src/main/java/uk/gov/hmcts/reform/hmc/api/services/RefDataServiceImpl.java

```diff
@@ -0,0 +1,88 @@
+package uk.gov.hmcts.reform.hmc.api.services;
+
+import feign.FeignException;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.HttpServerErrorException;
+import uk.gov.hmcts.reform.authorisation.generators.AuthTokenGenerator;
+import uk.gov.hmcts.reform.hmc.api.config.UserAuthTokenGenerator;
+import uk.gov.hmcts.reform.hmc.api.exceptions.RefDataException;
+import uk.gov.hmcts.reform.hmc.api.model.request.Hearing;
+import uk.gov.hmcts.reform.hmc.api.model.request.HearingUpdate;
+import uk.gov.hmcts.reform.hmc.api.model.response.CourtDetail;
+
+@Service
+@SuppressWarnings("unchecked")
+public class RefDataServiceImpl implements RefDataService {
+
+    @Autowired AuthTokenGenerator authTokenGenerator;
+    @Autowired UserAuthTokenGenerator userAuthTokenGenerator;
+    @Autowired RefDataApi refDataApi;
+    private static final Logger LOG = LoggerFactory.getLogger(RefDataServiceImpl.class);
+
+    /**
+     * This method will get all the court details of a particular venueId(epimmsId).
+     *
+     * @param epimmsId data to get court details from refData.
+     * @return courtDetail, particular Court detail.
+     */
+    @Override
+    @SuppressWarnings("unused")
+    public CourtDetail getCourtDetails(String epimmsId) {
+        CourtDetail courtDetail = null;
+        LOG.info("calling getCourtDetails service " + epimmsId);
+        try {
+            List<CourtDetail> courtDetailList =
+                    refDataApi.getCourtDetails(
+                            userAuthTokenGenerator.getSecurityTokens(),
+                            authTokenGenerator.generate(),
+                            epimmsId);
+            LOG.info("RefData call completed successfully" + courtDetailList);
+            List<CourtDetail> filteredCourtDetail =
+                    courtDetailList.stream()
+                            .filter(e -> "18".equals(e.getCourtTypeId()))
+                            .collect(Collectors.toList());
+            if (!filteredCourtDetail.isEmpty()) {
+                courtDetail = filteredCourtDetail.get(0);
```

Comment:
only one record is fine if come more than one court details for court type??

Reply:
For that court_type =18 only one record will come